spec/monads/optional_spec.rb

```diff
@@ -89,11 +89,24 @@ module Monads
       it 'forwards any unrecognised message to the value' do
         expect(value).to receive(:challenge)
         optional.challenge
+        expect { optional.method(:challenge) }.not_to raise_error
+        expect(optional.respond_to?(:challenge)).to be_truthy
```

Comment:
Stretch goal: the idiomatic RSpec is `expect(optional).to respond_to(:challenge)`.